runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -119,12 +119,13 @@ const CONSENSUS_WITHDRAW_HANDLER: &str = "consensus.WithdrawIntoRuntime";
 impl<Accounts: modules::accounts::API, Consensus: modules::consensus::API> API
     for Module<Accounts, Consensus>
 {
-    fn deposit<C: Context>(
+    fn deposit<C: TxContext>(
         ctx: &mut C,
         from: Address,
         amount: token::BaseUnits,
     ) -> Result<(), Error> {
         if ctx.is_check_only() {
+            Core::add_weight(ctx, TransactionWeight::ConsensusMessages, 1)?;
```

Comment:
Also move these outside of `is_check_only` branch to be consistent.

Reply:
hm but in that case this will emit 2 times when not in check tx (once here, and once in the consensus method invocation)

Reply:
Oh I see, ok. Maybe then just add a comment why this is so.

Reply:
makes sense, will add